docs/standard/serialization/system-text-json-source-generation-modes.md

```diff
@@ -0,0 +1,152 @@
+ms.date: 10/18/2021
+no-loc: [System.Text.Json]
+zone_pivot_groups: dotnet-version
+helpviewer_keywords:
+  - "JSON serialization"
+  - "serializing objects"
+  - "serialization"
+  - "objects, serializing"
+ms.topic: how-to
+---
+
+# How to choose reflection or source generation in System.Text.Json
+
+:::zone pivot="dotnet-6-0"
+> [!IMPORTANT]
+> Some information relates to prerelease product that may be substantially modified before it's released. Microsoft makes no warranties, express or implied, with respect to the information provided here.
+
+By default, `System.Text.Json` uses run-time reflection to gather the metadata it needs to access properties of objects for serialization and deserialization. As an alternative, `System.Text.Json` 6.0 can use the C# [source generation](../../csharp/roslyn-sdk/source-generators-overview.md) feature to improve performance, reduce private memory usage, and facilitate [assembly trimming](../../core/deploying/trimming/trim-self-contained.md), which reduces app size.
+
+You can use version 6.0 of System.Text.Json in projects that target earlier frameworks. For more information, see [How to get the library](system-text-json-overview.md#how-to-get-the-library).
+
+This article explains the options and provides guidance on how to choose the best approach for your scenario.
+:::zone-end
+
+:::zone pivot="dotnet-5-0,dotnet-core-3-1"
+[System.Text.Json](system-text-json-overview.md) version 6.0 and later can use the C# [source generation](../../csharp/roslyn-sdk/source-generators-overview.md) feature to improve performance, reduce private memory usage, and improve [assembly trimming](../../core/deploying/trimming/trim-self-contained.md) accuracy. You can use version 6.0 of System.Text.Json in projects that target earlier frameworks. For more information, see:
+
+* [How to get the library](system-text-json-overview.md#how-to-get-the-library)
+* [The .NET 6.0 version of this article](system-text-json-source-generation-modes.md?pivots=dotnet-6-0).
+
+:::zone-end
+
+:::zone pivot="dotnet-6-0"
+
+## Overview
+
+Choose reflection or source generation modes based on the following benefits that each one offers:
+
+| Benefit                                              | Reflection | Source generation:<br/>Metadata collection | Source generation:<br/>Serialization optimization |
+|------------------------------------------------------|------------|---------------------|----------------------------|
+| Simpler to code and debug.                           | ✔️        | ❌                  | ❌                        |
+| Supports non-public accessors.                       | ✔️        | ❌                  | ❌                        |
+| Supports init-only properties.                       | ✔️        | ❌                  | ❌                        |
+| Supports all available serialization customizations. | ✔️        | ❌                  | ❌                        |
```

Comment:
I believe metadata collection supports all customizations. Which ones did you find missing?